src/csi-wrapper/pkg/wrapper/nodeservice.go

```diff
@@ -125,10 +126,17 @@ func (s *NodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		podUid, volumeName := s.getPodUIDandVolumeName(targetPath)
 		savedPeerpodvolume.Labels["podUid"] = podUid
 		savedPeerpodvolume.Spec.PodUid = podUid
-		if volumeName != savedPeerpodvolume.Labels["volumeName"] {
-			glog.Error("The volume name from target path doesn't match with the CRD")
-			return
+		savedVolumeName := savedPeerpodvolume.Spec.VolumeName
+		if volumeName != savedVolumeName && savedVolumeName != peerpodVolumeNamePlaceholder {
+			glog.Error("The volume name from target path doesn't match with the CR")
+			return nil, errors.New("the volume name from target path doesn't match with the CR")
+		}
```

Comment:
In the original code, this function returns `nil, nil` when `volumeName != savedVolumeName`. The node returns `nil, error`. Do you think this is a bug in the the original code, and do you fix the bug in the new code?

Reply:
I think this was a bug in the previous version.
Not returning an error here makes it a bit hard to debug this case without checking the csi-wrapper logs, since it will result in simply never updating the peerpod volume CR, causing the Pod to be stuckin the creating phase